trucks.js

```diff
@@ -1,511 +1,604 @@
-    Twitter: 'https://twitter.com/fruitchatterbox'
+    Twitter: 'https://twitter.com/fruitchatterbox',
   },
   {
     name: 'Full Tilt Ice Cream',
-    type: ['Sweets'],
-    payment: ['Cash', 'Cards'],
+    type: ['Sweets',],
+    payment: ['Cash', 'Cards',],
     description: 'Handmade ice creams!',
     website: 'http://www.fulltilticecream.com',
     Facebook: 'https://www.facebook.com/fulltilticecreams',
-    Twitter: 'https://twitter.com/fticecream'
-  }
+    Twitter: 'https://twitter.com/fticecream',
+  },
 ];
 
 // this module should support the following methods:
 // getTrucks() - return all trucks
 // getTruck(name) - return the truck object matching 'name'
 // getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
 // filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
-// filterByFoodType(foodType) - return trucks with associated 'foodType'
\ No newline at end of file
+// filterByFoodType(foodType) - return trucks with associated 'foodType'
+
+/* eslint-enable max-len */
+/* eslint-disable no-console */
+/* eslint-disable no-undef */
+module.exports = {
+
+  /* eslint-enable no-undef */
+
+  getTrucks: function () {
+    return foodTrucks;
+  },
+
+  getTruck: function (truck) {
+    var foundTruck = 0;
+
+    for (var i = 0; i < foodTrucks.length; i++) {
+      if (foodTrucks[i].name === truck) {
+        var gotTruck = foodTrucks[i];
+        // console.log(foodTrucks[i].name);
+        foundTruck = 1;
+
+        return gotTruck;
+      }
+
+    }
+
+    if (foundTruck === 0) {
+      return (truck + ' truck not found');
+    }
+
+  },
+
+  getFoodTypes: function () {
+    var types;
+    for (var i = 0; i < foodTrucks.length; i++) {
+      types += foodTrucks[i].type + ' -> ' + foodTrucks[i].name + '\n';
+    }
+
+    return types;
+  },
+
+  filterTrucksByDay: function(day) {
+    if (day) {
+      var dayToCheck = day;
+    }
+
+    var trucksToday = [];
+    console.log('Day to check ' + dayToCheck);
+    for (var i = 0; i < foodTrucks.length; i++) {
+      var truckDays = [];
+      truckDays = foodTrucks[i].schedule;
+
+      if (truckDays) {
+        for (var j = 0; j < truckDays.length; j++) {
+          if (truckDays[j] === dayToCheck) {
+            // var gotTruck = foodTrucks[i];
+            // push to an array returns Object[object]
+            // return gotTruck;
+            trucksToday += foodTrucks[i].name + '\n';
+            // trucksToday += gotTruck;
+          }
+        }
+      }
+    }
+    /* eslint-disable newline-before-return */
+
+    return trucksToday;
+  },
+
+  filterTrucksByFoodType: function (foodType) {
+    var foundType = 0;
+    foodType = foodType.charAt(0).toUpperCase() + foodType.slice(1);
+    console.log('Looking for: ' + foodType);
+
+    for (var i = 0; i < foodTrucks.length; i++) {
+      if (foodTrucks[i].type == foodType) {  // why doesn't === work here?
```

Comment:
The `type` property here is an array. You don't want to compare a value against the entire array. Instead, use indexOf on the array to see if the value you're looking for is found in the array (see MDN for documentation of how array.indexOf works)